README.md

```diff
@@ -1,244 +1,90 @@
-
-```python
-from lakefs_spec import LakeFSFileSystem
-
-fs = LakeFSFileSystem()
-
-# The default is precheck=True, you can force the operation by setting precheck=False.
-fs.get_file("my-repo/my-ref/file.txt", "file.txt", precheck=True)
-```
-
-### Creating lakeFS automations in Python with `LakeFSFileSystem` hooks
-
-LakeFS has a variety of administrative APIs available through its Python client library.
-Within `lakefs-spec`, you can register hooks to your `LakeFSFileSystem` to run code after file system operations.
-A hook needs to have the signature `(client, context) -> None`, where the `client` argument holds the
-file system's lakeFS API client, and the `context` object contains information about the requested resource (repository, ref/branch, name).
-
-As an example, the following snippet installs a lakeFS hook that creates a commit on the lakeFS branch after a file upload:
-
-```python
-from lakefs_sdk.client import LakeFSClient
+from pathlib import Path
 
 from lakefs_spec import LakeFSFileSystem
 from lakefs_spec.client_helpers import commit
-from lakefs_spec.hooks import FSEvent, HookContext
-
-def create_commit_on_put(client: LakeFSClient, ctx: HookContext) -> None:
-    message = f"Add file {ctx.resource}"
-    commit(client, repository=ctx.repository, branch=ctx.ref, message=message)
-
-fs = LakeFSFileSystem()
-
-fs.register_hook(FSEvent.PUT_FILE, create_commit_on_put)
-
-# creates a commit with the message "Add file my-file.txt" after the file put.
-fs.put_file("my-file.txt", "my-repo/my-branch/my-file.txt")
-```
 
-### Implicit initialization and instance caching
-
-Aside from explicit initialization, you can also use environment variables and a configuration file (by default `~/.lakectl.yaml`) to initialize a lakeFS file system.
-The environment variables for the lakeFS client arguments are the names of the constructor arguments prefixed with `LAKEFS_`:
-
-```python
-import os
-from lakefs_spec import LakeFSFileSystem
-
-os.environ["LAKEFS_HOST"] = "localhost:8000"
-os.environ["LAKEFS_USERNAME"] = "username"
-os.environ["LAKEFS_PASSWORD"] = "password"
-
-fs = LakeFSFileSystem()
-```
-
-To initialize the lakeFS file system from a `lakectl` YAML configuration file, you can specify the `configfile` argument.
-
-```python
-from lakefs_spec import LakeFSFileSystem
+REPO, BRANCH = "repo", "main"
 
-# No argument means the default config (~/.lakectl.yaml) will be used.
-fs = LakeFSFileSystem(configfile="path/to/my/lakectl.yaml")
-```
+# Prepare example local data
+local_path = Path("demo.txt")
+local_path.write_text("Hello, lakeFS!")
 
-⚠️ To be able to read settings from a YAML configuration file, `pyyaml` has to be installed, for example using `pip`:
+# Upload to lakeFS and create a commit
+fs = LakeFSFileSystem()  # will auto-discover config from ~/.lakectl.yaml
+repo_path = f"{REPO}/{BRANCH}/{local_path.name}"
+fs.put(str(local_path), repo_path)
+commit(fs.client, REPO, BRANCH, "Add demo data")
```

Comment:
Great stuff! Makes me wonder if this is easier than the hook abstraction? I love this workflow more than the hook registration tbh (we should ask our users!).

Maybe we can explore transactions as a way of naturally creating commits? fsspec supports them, and it fits the natural versioning flow _very_ well in my opinion.

Reply:
It sits orthogonal to the hooks, IMO. Hooks I would see as the lower-level abstraction, in case you need to do something whenever some event occurs (like notifications, etc.), whereas transactions (or auto-commit mode #128) would allow a more convenient high-level way of sequencing the operations. Both have their valid place, but I think we should put more emphasis on the higher-level abstractions in the long run, since that's part of our USP.

Reply:
Okay, but are there enough interesting applications of hooks left if we decide to include support for versioning operations on transactions (i.e. on the bulked files)?

The more I think about it, implementing automatic commits, tagging, etc. in transactions and then directing users towards that is a solid way to go. How many people will be left that require event hook support then?

Also, the transaction concept is probably very much in line with the proposed lakeFS plans, and could be swapped out later with the SDK wrapper function if needed.

Reply:
Resolving this thread in favor of the discussions on the other related issues and PRs.